app/src/main/java/com/example/habitsmasher/PublicPrivateButtons.java

```diff
@@ -76,52 +84,62 @@ public PublicPrivateButtons(View view, boolean isPublic){
      * only one button may be clicked at a time.
      */
     public void setClickListeners(){
-        // on click listeners and logic behind the buttons switching
-        // TODO: Check interactions when the selected button is selected again
-        // preferably make it unselectable
-
+        // track whether the button is selected or not
         _publicButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
                 // if we check the public box, uncheck the private box
-                if (_privateSelected){
-                    _privateButton.performClick();
-                    _privateSelected = false;
+                if (_publicSelected){
+                    _publicSelected = false;
+                }
```

Comment:
I'm having trouble following this conditional statement. Why do we need to set `_publicSelected` to false if it is true? I was imagining something like: if public is selected, private selected is false.

Could you explain this a bit?

Reply:
I've gone back and changed the logic for the buttons, now both can be pressed or not pressed at the same time. The new checks in the validator reflects the constraints of the button, where only one of the two may be checked.